Gretty 3.0.8+ is incompatible with Proguard `7.3.0`

`build.gradle`:
```gradle
plugins {
	id 'org.gretty' version '3.0.8' apply false
}
```

`settings.gradle`:
```gradle
include 'foo'
```

`foo/build.gradle`:
```gradle
import proguard.gradle.ProGuardTask

buildscript {
	repositories {
		mavenCentral()
	}

	dependencies {
		classpath 'com.guardsquare:proguard-gradle:7.3.0'
	}
}

tasks.register('proguardFoo', ProGuardTask)
```

Running `gradle foo:proguardFoo` (with Gradle 7.5.1) on this fails with:

```
> Task :foo:proguardFoo FAILED

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':foo:proguardFoo'.
> Could not initialize class org.apache.logging.log4j.util.PropertiesUtil
```

Downgrading `ProGuard` to `7.2.2` or downgrading Gretty to `3.0.7` fixes the issue (i.e. there is another exception `Index 0 out of bounds for length 0` which is normal however).

Initially I thought that this was because of the recent changes to classloader filtering (as I'm using Gretty 3.1.0 and was testing with it). However, the fact that Gretty 3.0.8 also doesn't work... is extremely strange. Also note that Gretty is not even applied here but without this line, it works correctly.

@f4lco, any ideas?

--- a/libs/gretty-core/build.gradle
+++ b/libs/gretty-core/build.gradle
@@ -14,7 +14,6 @@ dependencies {
   api "commons-io:commons-io:$commons_io_version"
   api 'org.apache.commons:commons-lang3:3.3.2'
   api 'org.bouncycastle:bcprov-jdk15on:1.60'
-  api "org.springframework.boot:spring-boot-devtools:$springBootVersion"
   api project(':libs:gretty-common')
 }
 

--- a/libs/gretty-core/src/main/groovy/org/akhikhl/gretty/ServerConfig.groovy
+++ b/libs/gretty-core/src/main/groovy/org/akhikhl/gretty/ServerConfig.groovy
@@ -65,8 +65,6 @@ class ServerConfig {
 
   String portPropertiesFileName
 
-  Boolean liveReloadEnabled
-
   static ServerConfig getDefaultServerConfig(String serverName) {
     ServerConfig result = new ServerConfig()
     result.jvmArgs = []
@@ -79,7 +77,6 @@ class ServerConfig {
     result.redeployMode = 'restart'
     result.scanner = 'jetty'
     result.portPropertiesFileName = 'gretty_ports.properties'
-    result.liveReloadEnabled = false
     return result
   }
 

--- a/libs/gretty/src/main/groovy/org/akhikhl/gretty/DefaultLauncher.groovy
+++ b/libs/gretty/src/main/groovy/org/akhikhl/gretty/DefaultLauncher.groovy
@@ -14,7 +14,6 @@ import org.gradle.api.Project
 import org.gradle.process.JavaExecSpec
 import org.gradle.tooling.GradleConnector
 import org.gradle.tooling.ProjectConnection
-import org.springframework.boot.devtools.autoconfigure.OptionalLiveReloadServer
 
 import java.util.concurrent.Future
 
@@ -39,8 +38,6 @@ class DefaultLauncher extends LauncherBase {
 
   ScannerManager scannerManager
 
-  OptionalLiveReloadServer optionalLiveReloadServer
-
   DefaultLauncher(Project project, LauncherConfig config) {
     super(config)
     this.project = project
@@ -61,23 +58,6 @@ class DefaultLauncher extends LauncherBase {
     super.beforeJavaExec()
     scannerManager?.startScanner()
     project.file("${project.buildDir}/gretty_ports")
-    optionalLiveReloadServer?.startServer()
-    //
-    if(optionalLiveReloadServer) {
-      scannerManager?.registerFastReloadCallbacks((Closure) null, { -> optionalLiveReloadServer.triggerReload() })
-      //
-      Future response
-      scannerManager?.registerRestartCallbacks(
-              { -> response = reader.readMessageAsync() },
-              { -> response.get(); optionalLiveReloadServer.triggerReload() }
-      )
-      //
-      Future response2
-      scannerManager?.registerReloadCallbacks(
-              { -> response2 = reader.readMessageAsync() },
-              { -> response2.get(); optionalLiveReloadServer.triggerReload() }
-      )
-    }
   }
 
   protected File getPortPropertiesFile() {

--- a/libs/gretty/src/main/groovy/org/akhikhl/gretty/StartBaseTask.groovy
+++ b/libs/gretty/src/main/groovy/org/akhikhl/gretty/StartBaseTask.groovy
@@ -18,8 +18,6 @@ import org.gradle.api.tasks.Internal
 import org.gradle.api.tasks.TaskAction
 import org.gradle.process.JavaForkOptions
 import org.gradle.testing.jacoco.plugins.JacocoTaskExtension
-import org.springframework.boot.devtools.autoconfigure.OptionalLiveReloadServer
-import org.springframework.boot.devtools.livereload.LiveReloadServer
 /**
  * Base task for starting jetty
  *
@@ -54,10 +52,7 @@ abstract class StartBaseTask extends DefaultTask {
   void action() {
     LauncherConfig config = getLauncherConfig()
     Launcher launcher = new DefaultLauncher(project, config)
-    if(config.serverConfig.liveReloadEnabled) {
-      launcher.optionalLiveReloadServer = new OptionalLiveReloadServer(new LiveReloadServer())
-    }
-    launcher.scannerManager = createScannerManager(config, launcher.optionalLiveReloadServer)
+    launcher.scannerManager = createScannerManager(config)
     if(getIntegrationTest()) {
       boolean result = false
       try {
@@ -89,7 +84,7 @@ abstract class StartBaseTask extends DefaultTask {
     serverStartInfo = launcher.getServerStartInfo()
   }
 
-  private ScannerManager createScannerManager(LauncherConfig config, OptionalLiveReloadServer optionalLiveReloadServer) {
+  private ScannerManager createScannerManager(LauncherConfig config) {
     switch (config.serverConfig.scanner) {
       case 'jdk':
         if(JDKScannerManager.available()) {

--- a/libs/gretty/src/main/groovy/org/akhikhl/gretty/TaskWithServerConfig.groovy
+++ b/libs/gretty/src/main/groovy/org/akhikhl/gretty/TaskWithServerConfig.groovy
@@ -134,9 +134,6 @@ interface TaskWithServerConfig {
     @Optional @Input
     String getPortPropertiesFileName()
 
-    @Optional @Input
-    Boolean getLiveReloadEnabled()
-
     @ReplacedBy("serverConfigFile")
     Object getJettyXmlFile()
 

